src/System.Net.Http/src/System/Net/Http/Managed/HttpConnectionPools.cs

```diff
@@ -66,6 +77,12 @@ private void RemoveStalePools()
                 }
             }
 
+            // Stop running the timer if we don't have any pools to clean up.
+            if (_pools.IsEmpty)
+            {
+                _cleaningTimer.Change(Timeout.Infinite, Timeout.Infinite);
```

Comment:
Got it, yea I was also worried about the possibility of races here. Is it safe to lock on `_pools` to avoid the overhead of introducing a new sync object?